Kernel/Term.cpp

```diff
@@ -33,6 +33,15 @@ const unsigned Term::SF_FORMULA;
 const unsigned Term::SF_LAMBDA;
 const unsigned Term::SPECIAL_FUNCTOR_LOWER_BOUND;
 
+void Term::setId(unsigned id)
+{
+  CALL("Term::setId");
+  if (env.options->randomTraversals()) {
+    id += Random::getInteger(1 << 12) << 20; // the twelve most significant bits are randomized
```

Comment:
That's the idea! 

(Although, on a different branch, I was forced and able to get rid of `Term::id` completely so not sure how long this will last.)